driver/docker/docker.go

```diff
@@ -153,11 +158,41 @@ func (d *Driver) exec(op *driver.Operation) error {
 		AttachStdout: true,
 	}
 
-	hostCfg := &container.HostConfig{AutoRemove: true}
+	outputsFolder, err := ioutil.TempDir(d.config["OUTPUTS_MOUNT_PATH"], "outputs")
+	if err != nil {
+		return driver.OperationResult{}, err
+	}
+	defer os.RemoveAll(outputsFolder)
+
+	// On Mac, the default temp folder (set by the $TMPDIR env variable) is in /var.
+	// Docker for Mac (by default) can access these paths at /private/var but not /var.
+	// To make things work smoothly, we'll adjust the path when no config is set:
+	if runtime.GOOS == "darwin" {
+		if d.config["OUTPUTS_MOUNT_PATH"] == "" && strings.HasPrefix(outputsFolder, "/var") {
+			outputsFolder = "/private" + outputsFolder
+		}
+	}
+
+	err = os.Chmod(outputsFolder, 0777)
+	if err != nil {
+		return driver.OperationResult{}, err
+	}
+
+	hostCfg := &container.HostConfig{
```

Comment:
I wonder if we should use bind mounts, as it forbids executing the CNAB remotely 🤔 
I guess we should instead copy the output directory, symmetrically to the [inputs](https://github.com/deislabs/cnab-go/blob/master/driver/docker/docker.go#L178) which are tared then copied to the container (and not binded).

Reply:
The reason we chose to mount the directory was to ensure that the files would be available when the container stopped running. We've tried prototyping a version with `CopyFromContainer` subbed into the `fetchOutputs` implementation, but because the container has already stopped by that point, docker can't find the container to create the archive. Perhaps there's a way to restructure the control flow for [executing the container](https://github.com/deislabs/cnab-go/pull/82/files#diff-f7d55a687ffed0311363529637ca9749L221), but I don't think there's a point where you could know the outputs were ready but the container was also still available to read from.

If the driver needs to also support remote Docker servers, I think that's a follow up change someone could make after this PR where the code can use `volumes` instead and the user provides docker config to indicate the driver is not local.

Reply:
We can also:
-  remove the `autoremove` [here](https://github.com/deislabs/cnab-go/blob/master/driver/docker/docker.go#L156)
- defer the container removal
- wait for an `exited` status with the `WaitConditionNotRunning` condition [here](https://github.com/deislabs/cnab-go/blob/master/driver/docker/docker.go#L221)
- then use [CopyFromContainer](https://github.com/docker/cli/blob/f95ca8e1ba6c22c9abcdbf65e8dcc39c53958bba/vendor/github.com/docker/docker/client/interface.go#L76) on the `output` directory?

I guess this should work either locally or remotely ?

WDYT? Am I missing something?

Reply:
Thank you for clarifying what you meant! This does work. :)